File: app/src/main/java/com/github/kongpf8848/animation/evaluator/PointFEvaluator.java
package com.github.kongpf8848.animation.evaluator;

import android.animation.TypeEvaluator;
import android.graphics.Point;
import android.graphics.PointF;

/**
 * Created by pengf on 2017/3/23.
 */

public class PointFEvaluator implements TypeEvaluator<PointF> {



    public PointFEvaluator() {
    }

    @Override
    public PointF evaluate(float fraction, PointF startValue, PointF endValue) {
        float x=startValue.x+(endValue.x-startValue.x)*fraction;
        float y=startValue.y+(endValue.y-startValue.y)*fraction;
        return new PointF(x,y);

    }
}

File: app/src/main/java/com/github/kongpf8848/animation/evaluator/PointFEvaluator.kt
package com.github.kongpf8848.animation.evaluator

import android.animation.TypeEvaluator
import android.graphics.PointF

class PointFEvaluator : TypeEvaluator<PointF> {

    override fun evaluate(fraction: Float, startValue: PointF, endValue: PointF): PointF {
        val x = startValue.x + (endValue.x - startValue.x) * fraction
        val y = startValue.y + (endValue.y - startValue.y) * fraction
        return PointF(x, y)
    }
}